main.py

```diff
@@ -98,13 +98,21 @@ def main(repository: Path, output: Path, scantypes: tuple[str, ...]):
         )
 
 
-def make_archive(output: Path, repo_name: str, root: Path) -> Path:
+def make_archive(output: Path, repo_name: str, root: Path, directories: Path) -> Path:
     """
     create an archive of `directory` in `{output}/{repo_name}_timestamp.zip`
     """
-
     timestamp = datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S")
-    archive_name = f"{repo_name}_{timestamp}"
+
+    try :
+        linguist = Linguist()   
+        linguist_dir = directories.mkdir("linguist")
+        with open(f"{linguist_dir}/git-remotes") as git_remotes_file:
+            project_name = git_remotes_file.readline().split("/")[-1].split(".")[0]
```

Comment:
It would be better to use Python's built-ins to parse the path and extract the name we need, so that we handle cases like escaped charatcters etc. properly.

So if you add this import: `` and then do something like this:

```suggestion
            project_name = Path(urlparse(git_remotes_file.readline().split()[1]).path).stem
```

This works for both HTTPS and Git/SSH paths:
```
>>> Path(urlparse("https://github.com/Semalab/agent.git").path).stem
'agent'
>>> Path(urlparse("git@github.com:Semalab/smp-rest.git").path).stem
'smp-rest'
>>>
```